src/uct/sm/mm/posix/mm_posix.c

```diff
@@ -549,16 +549,29 @@ static ucs_status_t uct_posix_free(void *address, uct_mm_id_t mm_id, size_t leng
     return status;
 }
 
-static uct_mm_mapper_ops_t uct_posix_mapper_ops = {
-   .query   = ucs_empty_function_return_success,
-   .get_path_size = uct_posix_get_path_size,
-   .get_priority = uct_posix_get_priority,
-   .reg     = NULL,
-   .dereg   = NULL,
-   .alloc   = uct_posix_alloc,
-   .attach  = uct_posix_attach,
-   .detach  = uct_posix_detach,
-   .free    = uct_posix_free
+static uct_mm_md_mapper_ops_t uct_posix_md_ops = {
+   .super = {
+        .close                  = uct_mm_md_close,
+        .query                  = uct_mm_md_query,
+        .mem_alloc              = uct_mm_mem_alloc,
+        .mem_free               = uct_mm_mem_free,
+        .mem_advise             = (void*)ucs_empty_function_return_unsupported,
+        .mem_reg                = (void*)ucs_empty_function_return_unsupported,
+        .mem_dereg              = (void*)ucs_empty_function_return_unsupported,
+        .mkey_pack              = uct_mm_mkey_pack,
+        .is_sockaddr_accessible = (void*)ucs_empty_function_return_zero,
+        .detect_memory_type     = (void*)ucs_empty_function_return_unsupported
```

Comment:
it seems that it could break PGI compilation. 
I think the approach from https://github.com/openucx/ucx/commit/4d1a87dfd9950aa4698fd12fa0fe74bed06f51cd should be done here:
```
.mem_advise             = (uct_md_mem_advise_func_t)ucs_empty_function_return_unsupported,
```
```
.mem_reg                = (uct_md_mem_reg_func_t)ucs_empty_function_return_unsupported,
```
```
.mem_dereg              = (uct_md_mem_dereg_func_t)ucs_empty_function_return_unsupported,
```
```
.is_sockaddr_accessible = (uct_md_is_sockaddr_accessible_func_t)ucs_empty_function_return_zero,
```
```
.detect_memory_type     = (uct_md_detect_memory_type_func_t)ucs_empty_function_return_unsupported
```
and the same below